sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -288,33 +286,41 @@ private static void CloseLink(RequestResponseAmqpLink link)
                 }
                 cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
-                var messagesReceived = await Task.Factory.FromAsync
+                var messagesReceived = await Task.Factory
+                    .FromAsync<(ReceivingAmqpLink, int, TimeSpan?, TimeSpan), IEnumerable<AmqpMessage>>
                 (
-                    (callback, state) => link.BeginReceiveRemoteMessages(
-                        maxMessages,
-                        TimeSpan.FromMilliseconds(20),
-                        maxWaitTime ?? timeout,
-                        callback,
-                        state),
-                    (asyncResult) => link.EndReceiveMessages(asyncResult, out amqpMessages),
-                    TaskCreationOptions.RunContinuationsAsynchronously
-                ).ConfigureAwait(false);
+                    static (arguments, callback, state) =>
+                    {
+                        var (link, maxMessages, maxWaitTime, timeout) = arguments;
+                        return link.BeginReceiveRemoteMessages(
+                            maxMessages,
+                            TimeSpan.FromMilliseconds(20),
```

Comment:
@JoshLove-msft:  Is this the potential `TimeSpan.Zero` change that has been discussed, or is this an unrelated batching timeout?